snapcraft/internal/sources/_base.py

```diff
@@ -37,6 +37,7 @@ def __init__(self, source, source_dir, source_tag=None, source_commit=None,
         self.source_branch = source_branch
         self.source_depth = source_depth
         self.source_checksum = source_checksum
+        self.assets = None
```

Comment:
`assets` is a bad name, it is a source asset for snapcraft itself, but for the source class in question it is not an asset.